src/pages/RegistrationPage/RegistrationPage.tsx

```diff
@@ -0,0 +1,152 @@
+import { Flex, useDisclosure } from '@chakra-ui/react';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useEffect, useState } from 'react';
+import { useForm, UseFormRegister } from 'react-hook-form';
+import { useNavigate } from 'react-router';
+import { ZodType } from 'zod';
+
+import { FullScreenSpinner } from '~/components/FullScreenSpinner/FullScreenSpinner';
+import { AccountRecoveryType } from '~/pages/LoginPage/PasswordRecovery/RecoveryPasswordModal/Steps/RessetPassword';
+import { RegisterSuccessModal } from '~/pages/RegistrationPage/Modals/RegisterSuccessModal';
+import { VerificationErrorModal } from '~/pages/RegistrationPage/Modals/VerificationErrorModal';
+import { RegistrationButtons } from '~/pages/RegistrationPage/RegistrationButtons/RegistrationButtons';
+import { RegistrationProgress } from '~/pages/RegistrationPage/RegistrationProgress/RegistrationProgress';
+import { FirstStepInputs } from '~/pages/RegistrationPage/Steps/FirstStepInputs/FirstStepInputs';
+import { SecondStepInputs } from '~/pages/RegistrationPage/Steps/SecondStepInputs/SecondStepInputs';
+import { useSignUpMutation } from '~/query/services/auth';
+import { ErrorResponse } from '~/query/types/types';
+import { EMAIL_EXISTS, LOGIN_EXISTS } from '~/shared/constants/authStatuses';
+import { DATA_TEST_IDS } from '~/shared/constants/dataTestIds';
+import { useAlertToast } from '~/shared/hooks/useAlertToast';
+import { AuthLayout } from '~/shared/layouts/AuthLayout/AuthLayout';
+import { userDataSchema, userPasswordSchema } from '~/shared/types/validationSchemas/signUpSchema';
+import { emailVerifiedSelector } from '~/store/app-slice';
+import { useAppSelector } from '~/store/hooks';
+
+export type RegisterFormDataType = {
+    firstName: string;
+    lastName: string;
+    email: string;
+    login: string;
+    password: string;
+    passwordConfirm: string;
+};
+
+const signUpSchema: ZodType[] = [userDataSchema, userPasswordSchema];
+
+export const RegistrationPage = () => {
+    const [step, setStep] = useState(0);
+    const [tempData, setTempData] = useState<RegisterFormDataType>();
+    const [email, setEmail] = useState('');
+    const isVerified = useAppSelector(emailVerifiedSelector);
+
+    const alertToast = useAlertToast();
+    const navigate = useNavigate();
+
+    useEffect(() => {
+        if (isVerified === false) {
```

Comment:
зачем тут сравнение?
```suggestion
        if (!isVerified) {
```

Reply:
потому что там изначально состояние null. false ставиться только если перейти с 'verification?emailVerified=false' . С таким вариантом if (!isVerified)  условие будет тру всегда и будет появляться модалка при любом переходе на /register